changelog/@unreleased/pr-791.v2.yml

```diff
@@ -0,0 +1,8 @@
+type: improvement
+improvement:
+  description: Compile with `-Werror` and `-Xlint:deprecation` to catch deprecations
```

Comment:
Could you describe the change in a way that would be helpful for consumers of the plugin? something like: `gradle-baseline should now be compatible with Gradle  6.0. gradle-baseline now requires at least Gradle 5.0 and is no longer compatible with 4.X.`

Reply:
I am not sure we are compatible with gradle 6 though since it's not released. I could say it's compatible with gradle snapshot that I tested with.

Reply:
Also if I was just making it compatible with gradle 6 I wouldn't be adding the compiler flags

Reply:
Yeah I'd just phrase this as a break and declare compatibility with `6.0-20190904072820+0000`

Reply:
Mmmm looks like there are more stragglers outside Foundry, but I think this is probably OK:

![image](https://user-images.githubusercontent.com/3473798/64268198-91521900-cf2f-11e9-9aa0-cd701be9cd0c.png)